### ISSUE TITLE
Unexpected physics behavior when modifying position

### ISSUE THREAD

**Issue by lypoluz** [godotengine/godot#88376]
### Tested versions

reproducable in
v4.3.dev3.mono.official [36e943b6b]
**v4.2.1.stable.mono.official [b09f793f5]**
v4.2.1.stable.official [b09f793f5]
v4.1.4.rc1.mono.official [b9008f3d5]
v4.0.stable.mono.official [92bee43ad]

### System information

Godot v4.2.1.stable.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3070 (NVIDIA; 31.0.15.5123) - Intel(R) Core(TM) i7-9800X CPU @ 3.80GHz (16 Threads)

### Issue description

Currently, using a CharacterBody3D as a platform with a RigidBody3D on top, when setting the position (`postion += ...`) of the platform, the RigidBody moves along. What i would expect is that the RigidBody stays in place.  This is especially notable when doing many small steps. When stepping by a large amount, the RigidBody also moves slightly.
This video demonstrates the described interaction.

https://github.com/godotengine/godot/assets/37896314/d86677bb-4ed6-4360-8fa8-7d7c54cdf0ee

They interact similar when using `global_transform.origin += ...` or `move_and_collide(...)`.
I would expect this behavior when using `move_and_slide()` or `move_and_collide(...)` but not when setting the position directly.

Additional info:
I tried reproducing the conveyor behavior shown in this video: https://www.youtube.com/watch?v=hC1QZ0h4oco
This bug might be related to https://github.com/godotengine/godot/issues/21560

### Steps to reproduce

platform (CharacterBody3D):
![image](https://github.com/godotengine/godot/assets/37896314/648c6163-22bc-49de-aea0-baac6917a1c8)
With script on CharacterBody3D:
```
extends CharacterBody3D
var speed = 1
func _physics_process(delta):
    position.x += delta * speed
    # similar interaction
    # global_transform.origin += delta * speed
    # move_and_collide(Vector3(1,0,0) * delta * speed)
```

movable object (RigidBody3D):
![image](https://github.com/godotengine/godot/assets/37896314/e6ff72fb-88fb-48d0-84ef-8b51f0d7d2df)

Let the RigidBody fall on the CharacterBody and observe.

### Minimal reproduction project (MRP)

The MRP includes a bool for using bigger steps. The video was made using this MRP.
[test_conveyor.zip](https://github.com/godotengine/godot/files/14301438/test_conveyor.zip)

**Comment by ryevdokimov** [comment]
You might be interested in my project: [Open Industry Project](https://github.com/Open-Industry-Project/Open-Industry-Project). Feel free to look through the code, although it's in C#. 

Long story short though. You don't want to teleport the position of colliders for the purpose of moving things with it. It's going to bypass the physics engine. You want to kind of do the reverse. I would recommend using a RigidBody3D, modifying the linear velocity and then teleporting the conveyor back in place.

**Comment by lypoluz** [comment]
> You might be interested in my project: [Open Industry Project](https://github.com/Open-Industry-Project/Open-Industry-Project). Feel free to look through the code, although it's in C#.
> 
> Long story short though. You don't want to teleport the position of colliders for the purpose of moving things with it. It's going to bypass the physics engine. You want to kind of do the reverse. I would recommend using a RigidBody3D, modifying the linear velocity and then teleporting the conveyor back in place.

@ryevdokimov
Thanks for your info and sharing your project. I am well aware of other ways to implement a conveyor. However, I like to experiment and try to find different solutions to the same problem/goal.

**Comment by ryevdokimov** [comment]
Fair enough. I understand the intention of the post now, and I agree with your assessment. It seems like it's NOT bypassing the physics engine as it should be.

**Comment by ryevdokimov** [comment]
I can confirm that the behavior of RigidBody3D and CharacterBody3D are different in this regard. When using a RigidBody3D the object on top will remain in place.
